### PR TITLE
Remove hacks to make llvmlite threadsafe

### PR DESCRIPTION
**Pull request by sklam**
(Don't merge until 0.36.0 is released)

Depends on https://github.com/numba/llvmlite/pull/318

### CODE REVIEW

File: numba/llvmthreadsafe.py [hunk 1]
```diff
@@ -1,68 +0,0 @@
-"""
```
Comment (stuartarchibald) [inline numba/llvmthreadsafe.py]: I am glad to see this gone!

File: numba/runtime/nrt.py [hunk 2]
```diff
@@ -16,7 +15,6 @@ class _Runtime(object):
     def __init__(self):
         self._init = False
 
-    @lock_llvm # prevent race to install compiled library functions
```
Comment (stuartarchibald) [inline numba/runtime/nrt.py]: I think a lock here is still needed, it was an abuse of the already existing `@lock_llvm` for the purposes of ensuring threadsafety on initialisation opposed to threadsafety in LLVM calls themselves. I think we need a more general initialisation lock anyway as I'm fairly sure that this https://github.com/numba/numba/issues/2517 is a result of a race condition during the initialisation sequence, this lock should be applied here too. For ref, this patch added the current lock and has the explanation for the addition https://github.com/numba/numba/pull/2449. Thoughts?

Comment (sklam) [reply numba/runtime/nrt.py]: I can borrow the `numba.lock.lock_compiler` here.

Comment (sklam) [reply numba/runtime/nrt.py]: See https://github.com/numba/numba/pull/2627/commits/bdab4f1bae897b1770591a7b4fa0faf7a32610d5

**Comment by anton-malakhov** [comment]
Is it fine if this PR is not applied (using numba 0.36.2) while using llvmlite 0.22 with https://github.com/numba/llvmlite/pull/318 ? Just double-locking or some deadlock might appear?

**Comment by seibert** [comment]
@sklam: Would that work?

**Comment by anton-malakhov** [comment]
Please ignore. It didn't help. I still see the crashes

**Comment by sklam** [comment]
There won't be any deadlock.  Just extra unnecessary locking.

**Pull request merged**